### PR TITLE
Remove `FieldExt` and `Group` traits

### LINKED ISSUE #42: Kill `FieldExt` trait
This (along with its other related traits) was a helper trait that we created to provide additional methods needed for `halo2_proofs` that the `ff` crate traits didn't have, while we figured out what those needed additional methods _were_. Now that we've cut `halo2_proofs 0.1.0`, we need to finish the job:
- Figure out how to better abstract these additional methods if necessary.
- Add these abstractions to `ff`.
- Delete the `FieldExt` trait.

### PR DESCRIPTION
**Pull request by str4d**
```
One, two! One, two! And through and through
    The vorpal blade went snicker-snack!
He left it dead, and with its head
    He went galumphing back.
```

Depends on https://github.com/zkcrypto/ff/pull/98.
Closes zcash/pasta_curves#42.

### CODE REVIEW

File: src/arithmetic/curves.rs [hunk 1]
```diff
@@ -28,12 +25,11 @@ pub trait CurveExt:
     + ConditionallySelectable
     + ConstantTimeEq
     + From<<Self as PrimeCurve>::Affine>
-    + Group<Scalar = <Self as group::Group>::Scalar>
 {
```
Comment (daira) [inline src/arithmetic/curves.rs:28]: On line 67, consider disabling the type_complexity lint.
```diff
@@ -68,6 +68,7 @@ pub trait CurveExt:
     ///     (g * x + &(h * r)).to_affine()
     /// }
     /// ```
+    #[allow(clippy::type_complexity)]
     fn hash_to_curve<'a>(domain_prefix: &'a str) -> Box<dyn Fn(&[u8]) -> Self + 'a>;
 
     /// Returns whether or not this element is on the curve; should
```

File: src/fields/fp.rs [hunk 2]
```diff
@@ -580,11 +563,30 @@ impl ff::Field for Fp {
...
     type Repr = [u8; 32];
 
+    const MODULUS: &'static str =
+        "0x40000000000000000000000000000000224698fc094cf91b992d30ed00000001";
+    const TWO_INV: Self = Fp::from_raw([
+        0xcc96987680000001,
+        0x11234c7e04a67c8d,
+        0x0000000000000000,
+        0x2000000000000000,
+    ]);
```
Comment (daira) [inline src/fields/fp.rs:573]: I have not checked any of the constants.

**Review by daira — APPROVED** [review]
utACK

File: src/fields/fp.rs (hunk 2 above)
Comment (str4d) [reply src/fields/fp.rs:573]: The constants should all be pure moves; no news constants were calculated here.

File: src/arithmetic/curves.rs (hunk 1 above)
Comment (str4d) [reply src/arithmetic/curves.rs:28]: I'll do these cleanups in a separate PR.

Comment (daira) [reply src/arithmetic/curves.rs:28]: I have clippy cleanups locally, so I can make that PR if you want.

**Review by daira — APPROVED** [review]
utACK; also checked that the constants are only moved.

**Review by ebfull — APPROVED** [review]

**Pull request merged**